cmd/soroban-rpc/internal/methods/get_ledger_entries.go

```diff
@@ -47,6 +48,14 @@ func NewGetLedgerEntriesHandler(logger *log.Entry, ledgerEntryReader db.LedgerEn
 		var ledgerKeys []xdr.LedgerKey
 		for i, requestKey := range request.Keys {
 			var ledgerKey xdr.LedgerKey
+			if err := gxdr.ValidateLedgerKey(requestKey, gxdr.DefaultMaxDepth); err != nil {
+				logger.WithError(err).WithField("request", request).
+					Infof("could not validate ledgerKey at index %d from getLedgerEntries request", i)
+				return GetLedgerEntriesResponse{}, &jrpc2.Error{
+					Code:    jrpc2.InvalidParams,
+					Message: fmt.Sprintf("cannot unmarshal key value %s at index %d", requestKey, i),
```

Comment:
I'm not sure that we should duplicate the ledger key in the response. The caller should know what's the "third" or "fifth" ledger key they sent in the request.